src/Workspaces/Core/Portable/Storage/SQLite/Interop/SafeSqliteBlobHandle.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using SQLitePCL;
+
+namespace Microsoft.CodeAnalysis.SQLite.Interop
+{
+    internal sealed class SafeSqliteBlobHandle : SafeHandle
+    {
+        private readonly sqlite3_blob? _wrapper;
+        private readonly SafeHandleLease _lease;
+
+        public SafeSqliteBlobHandle(SafeSqliteHandle sqliteHandle, sqlite3_blob? wrapper)
+            : base(invalidHandleValue: IntPtr.Zero, ownsHandle: true)
+        {
+            _wrapper = wrapper;
+            SetHandle(wrapper?.ptr ?? IntPtr.Zero);
+            _lease = sqliteHandle.Lease();
+        }
+
+        public override bool IsInvalid => handle == IntPtr.Zero;
+
+        public new sqlite3_blob DangerousGetHandle()
```

Comment:
FWIW, I was very confused by the code until I got to this.  I suggest picking a different method name.

Reply:
This works exactly the same way as `SafeHandle.DangerousGetHandle()` would have worked if `raw` took `IntPtr` instead of the intermediate wrapper types. I reused the name because all considerations for how to treat the handle when this method is used are the same considerations someone would need for `SafeHandle.DangerousGetHandle()`.

Reply:
You possibly also want `if (IsClosed) { throw new WhateverDangerousGetHandleThrowsException(); }` here.

Reply:
Will rename in the master branch.

> You possibly also want `if (IsClosed) ...` here

This was brought up in the initial review. I opted to use the same approach that `DangerousGetHandle` uses: as a `Dangerous*` method, it is the caller's sole responsibility to validate preconditions.